### PR TITLE
Add configurable layers in manifest events

### PR DESCRIPTION
**Pull request by andrew-leung**
Adds a configurable option to include layer info in manifest notification events. When we receive manifest notification events we often want to know which layers were accessed and/or compose the image. Rather than re-querying the registry, it is convenient to have that information in the manifest event itself.

With the change the default behavior of not including layer information in manifest events is not changed. An explicit configuration is needed to include the info.

### CODE REVIEW

**Comment by andrew-leung** [comment]
Hi @dmcgowan , could you take a look please or refer the PR to who should take a look? Thanks!

File: notifications/bridge.go [hunk 1]
```diff
@@ -12,11 +12,12 @@ import (
 )
 
 type bridge struct {
-	ub      URLBuilder
-	actor   ActorRecord
-	source  SourceRecord
-	request RequestRecord
-	sink    Sink
+	ub             URLBuilder
+	manfiestLayers bool
```
Comment (fabiokung) [inline notifications/bridge.go]: typo: s/manfiestLayers/manifestLayers/

**Comment by andrew-leung** [comment]
PTAL @dmp42 @dmcgowan @caervs, thanks!

**Comment by dmcgowan** [comment]
@manishtomar I agree `ManifestLayers`/`Layers` -> `References` is more consistent with the terminology we are _trying_ to use in the registry and especially any newly exposed interface.

**Comment by andrew-leung** [comment]
@manishtomar @dmcgowan , good suggestion. I've updated the terminology to use `References` over `Layers / ManifestLayers`.

**Comment by andrew-leung** [comment]
@manishtomar & @dmcgowan , please let me know if there is anything else needed prior to merging.

**Comment by dmcgowan** [comment]
LGTM

**Comment by manishtomar** [comment]
@andrew-leung Sorry lost track of this. This looks good from my side. @dmcgowan @dmp42 Please take another look and if looks good please merge it.

**Comment by stevvooe** [comment]
This change isn't great. It breaks the abstraction and pulls format specific items up to the event, which makes migration impossible. This will also making adding new formats much more challenging.

 Applications that need this information should just pull the manifest to get the references.

Also, this should have incremented the version for the event mediatype.

In the future, let's make sure we've exhausted all client-side options before adding new, format-specific functionality to the registry.

**Pull request merged**